minetest/minetest
# [ADOPTION] Meta get/set bool functions

This PR adds the missing get_bool and set_bool to the series of *meta:get_*/*meta:set_* functions.
The implementation mirrors the others.

This pr is also an adoption of #12894 
This is the same as that pr except. I ported to 5.10.0-dev and rebased.

- Goal of the PR
Add meta:set_bool and meta:get_bool
- How does the PR work?
It mirrors other functions to get the result

## To do

Ready for Review.

## How to test

Compile and ensure it works properly, then run ```/unittests```. To ensure there are no errors.

```
minetest.register_tool(":get_set_bool_test", {
  on_place = function(itemstack, player, pointed_thing)
    itemstack:get_meta():set_bool("mybool", true)
    return itemstack
  end,
  on_secondary_use = function(itemstack, player, pointed_thing)
    itemstack:get_meta():set_bool("mybool", false)
    return itemstack
  end,
  on_use = function(itemstack, player, pointed_thing)
    minetest.chat_send_player(player:get_player_name(),
      "Previously right clicked on node: " ..
      tostring(itemstack:get_meta():get_bool("mybool"))
    )
  end,
})
```
